### DISCUSSION TITLE
Metadata

### DISCUSSION THREAD

**Discussion by johnlindquist** [johnlindquist/kit#185] [Guide]
### `// Shortcut`

```js
// Shortcut: cmd opt shift ctrl a
```

Automatically assigns a keyboard shortcut to the script. Will update each time you save the file. You can also change shortcuts in `Main Menu -> Help -> Change script shortcut`

### `// Menu`

```js
// Menu: My Awesome Script
```

Customizes the name of the script in the main menu

### `// Description`

```js
// Description: This script does something awesome
```

Describes the script in the main menu

### `// Author` and `// Twitter`

Credits you when the script runs with a prompt

### `// Exclude`

```js
// Exclude: true
```

Hides the script from the Main Menu (can still be run from shortcuts, schedule, system, etc).

### `// Schedule`

```js
// Schedule: */10 * * * * *
```

Runs the script on a cron schedule.

### `// System`

```js
// System: unlock-screen
```

Runs the script on a system event. 

Available events:
* suspend
* resume
* on-ac
* on-battery
* shutdown
* lock-screen
* unlock-screen
* user-did-become-active
* user-did-resign-active

Read about the available events [here](https://www.electronjs.org/docs/api/power-monitor#events)

**Comment by peterjohnhunt** [comment]
Duuuude, these system triggers are an awesome addition! Also super excited for the exclude as i've broken out my scripts into a lot of subscripts that are made available in a "main" script with tabs, similar to how you have menus.

**Reply by peterjohnhunt** [reply]
Would custom order value be something you might add down the road as a metadata comment? I.e. if you want scripts to sort non alphabetically

**Reply by johnlindquist** [reply]
I'll make a post soon on how to "build your own main menu" with an example. Then you could map whatever shortcut you want to it and customize tabs/orders/whatever

**Reply by kenNash** [reply]
Wow, cool!

Did you do that?

**Reply by johnlindquist** [reply]
I haven't... The general idea is like this though:


[Open testing-new-main-menu in Script Kit](https://scriptkit.com/api/new?name=testing-new-main-menu&url=https://gist.githubusercontent.com/johnlindquist/a1bbc2ab06195056a457f420bf38f955/raw/b29dc44ea70492462f16f1883391d8a64419edfa/testing-new-main-menu.ts")

```js
// Name: Testing New Main Menu
// Shortcut: cmd .

import "@johnlindquist/kit"

onTab("Run", async (input = "") => {
    await attemptImport(kitPath("cli", "app-run.js"), "--input", input)
})

onTab("A", async (input = "") => {
    await selectScript("Scripts that start with 'A'", true, (scripts)=> scripts.filter(script => script.name.startsWith("a")))
})

onTab("B", async (input = "") => {
    await selectScript("Scripts that start with 'B'", true, (scripts)=> scripts.filter(script => script.name.startsWith("B")))
})
```

⭐️ P.S. - We're starting up a Discord server if you want to hang out and chat about script ideas/questions/whatever: https://discord.gg/pY84ERdK
